api/envoy/extensions/common/key_value/file_based/v3/config.proto

```diff
@@ -0,0 +1,27 @@
+syntax = "proto3";
+
+package envoy.extensions.common.key_value.file_based.v3;
```

Comment:
Nit: this should be in `api/envoy/extensions/key_value/filed_based/v3/config.proto` and same package namespace.

The rationale is that the `KeyValueConfig` is a common extension point declaration reused across protos. The `FileBasedKkeyValueStoreConfig` is a `key_value` extension, and all extension classes have a root like `api/envoy/extensions/<extension class>`.